### PR TITLE
Add CREF about ambiguous behaviour of additionalProperties

### LINKED ISSUE #1172: ambiguous behaviour of additionalProperties when invalid
I was testing a schema and getting an additionalProperties error that didn't make much sense to me as the name of the property was in the schema.

So I put together a smaller schema to reproduce it.

```json
{
    "type": "object",
    "properties": {
        "test": {
            "type": "string"
        }
    },
    "additionalProperties": false
}
```

I tried testing it against this piece of json.

```json
{
  "test": 1
}
```

I tried it against a few of the online validators and depending on which implementation I tried the errors that I would get are different, sometimes I would just get the error about `test` not being a string, other times I would also get an error about no additional properties.

Reading http://json-schema.org/understanding-json-schema/reference/object.html#additional-properties

Specifically around `properties whose names are not listed in the properties keyword or match any of the regular expressions in the patternProperties keyword` implies that only the property name is important, not if it's valid or not.

I know that some of these implementations use annotations in order to determine if a property is additional or not, and those annotations will get dropped if any properties are invalid as referenced here, which is probably why I'm seeing what I am.

https://github.com/json-schema-org/json-schema-spec/issues/939

Is there any expectations as to how additionalProperties is meant to work in these instances?

### PR DESCRIPTION
**Pull request by Relequestual**
Fixes #1172
Must see new issue relating to the behaviour of annotation collection for resolving in the next draft.
Related to https://github.com/json-schema-org/community/discussions/57

### CODE REVIEW

File: jsonschema-core.xml [hunk 1]
```diff
@@ -2480,6 +2480,28 @@
                             checking the names in "properties" and the patterns in
                             "patternProperties" against the instance property set.
                             Implementations that do not support annotation collection MUST do so.
+                            <cref>
+                                In defining this option, it seems there is the potential for
+                                ambiguity in the output format. The ambiguity does not effect validation results,
+                                but it does effect the resutling output format.
```
Comment (karenetheridge) [inline jsonschema-core.xml:2486]:
```suggestion
                                ambiguity in the output format. The ambiguity does not affect validation results,
                                but it does affect the resulting output format.
```

File: jsonschema-core.xml [hunk 2]
```diff
@@ -2480,6 +2480,28 @@
...
+                                The ambiguity allows for multiple valid output results depending on if annotations
+                                are used or a solution that "produces the same effect" as draft-07. It is understood
+                                that annotations from failing schemas are dropped. See
+                                https://github.com/json-schema-org/community/discussions/57 for details.
+                                We discussed the possible solutions at length (https://github.com/json-schema-org/json-schema-spec/issues/1172).
+                                We concluded that we need to re-evaluate the annotation collection system
+                                and its use for validation purposes, but we also agreed not for this release.
```
Comment (karenetheridge) [inline jsonschema-core.xml:2493]:
```suggestion
                                and its use for validation purposes, but we also agreed not to do so for this release.
```

File: jsonschema-core.xml [hunk 3]
```diff
@@ -2480,6 +2480,28 @@
                             checking the names in "properties" and the patterns in
                             "patternProperties" against the instance property set.
                             Implementations that do not support annotation collection MUST do so.
+                            <cref>
+                                In defining this option, it seems there is the potential for
+                                ambiguity in the output format. The ambiguity does not effect validation results,
+                                but it does effect the resutling output format.
+                                The ambiguity allows for multiple valid output results depending on if annotations
```
Comment (karenetheridge) [inline jsonschema-core.xml:2487]:
```suggestion
                                The ambiguity allows for multiple valid output results depending on whether annotations
```

File: jsonschema-core.xml [hunk 4]
```diff
@@ -2480,6 +2480,28 @@
...
+                                are used or a solution that "produces the same effect" as draft-07. It is understood
+                                that annotations from failing schemas are dropped. See
+                                https://github.com/json-schema-org/community/discussions/57 for details.
+                                We discussed the possible solutions at length (https://github.com/json-schema-org/json-schema-spec/issues/1172).
+                                We concluded that we need to re-evaluate the annotation collection system
+                                and its use for validation purposes, but we also agreed not for this release.
+                                This confusion stemmed from the understanding of "evaluated" and the impact on
+                                annotation collection.
+                                It was proposed that we could read a different interpritation (https://github.com/json-schema-org/json-schema-spec/issues/1172#issuecomment-1049686478)
+                                however the previous interpritation is still valid.
```
Comment (karenetheridge) [inline jsonschema-core.xml:2497]:
```suggestion
                                It was proposed that we could read a different interpretation (https://github.com/json-schema-org/json-schema-spec/issues/1172#issuecomment-1049686478)
                                however the previous interpretation is still valid.
```

**Review by karenetheridge — CHANGES_REQUESTED** [review]

**Comment by Relequestual** [comment]
I've removed some of the waffle from the CREF in the spec document as @jdesrosiers requested.
I've also added an Architectural Decision Record (ADR) for this overall decision to capture the reasoning behind it.
As you are named on the ADR (@karenetheridge @jdesrosiers @gregsdennis), it would be great to get your reviews.

Please feel free to make suggestion changes in your reviews if any additional thoughs come to mind or I've forgotten an option we considered.

**Review by karenetheridge — APPROVED** [review]
+1. Moving the decision content to an ADR is a good move as it allows more verbosity as well as linking to all the related issues where the discussions took place.

File: adr/2022-04-08-cref-for-ambiguity-and-fix-later-gh-spec-issue-1172.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+## Context and Problem Statement
+
+When we changed the specification to use annotations as the context in which some keywords behave, we included a clause that implementations which didn't use annotations may implement or optimize the processing of `additionalProperties` in another way which produces the same effect as the prior behaviour.
+This section created an ambiguity in terms of the resulting output format, but not validation.
+
+We needed to decide on how to proceed for the patch release of the 2020-12 version of the specification.
+
+The two above links are to a GitHub Discussion and a GitHub Issue detailing the problems.
+Details with an example of the problem can be see in the Discussion's opening post specifically.
```
Comment (jdesrosiers) [inline adr/2022-04-08-cref-for-ambiguity-and-fix-later-gh-spec-issue-1172.md:19]:
```suggestion
Details with an example of the problem can be seen in the Discussion's opening post specifically.
```

**Review by jdesrosiers — APPROVED** [review]
Looks good. Creating an ADR for this was a great idea! If we had one for when the annotation based behavior was was introduced, we could have saved ourselves a lot of speculation about what the intention was.

File: adr/2022-04-08-cref-for-ambiguity-and-fix-later-gh-spec-issue-1172.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+* The "patch release" should not change anything functionally
+* Annotations as they are, are confusing to users, implementers, and specification editors alike
+* Patch release is behind schedule
+* There are currently no tests for the output format
+* It's hard to see any immediate consensus on changing the annotation based behaviour
+
+## Considered Options
+
+* [Leaving it "as is" and do nothing](https://github.com/json-schema-org/community/discussions/57#discussioncomment-1413777)
+* [Pick one]()https://github.com/json-schema-org/community/discussions/57#discussioncomment-1416683) of the behaviours
```
Comment (gregsdennis) [inline adr/2022-04-08-cref-for-ambiguity-and-fix-later-gh-spec-issue-1172.md:32]: Link formatting

**Review by gregsdennis — APPROVED** [review]
Minor formatting issue. Good otherwise.

**Comment by Relequestual** [comment]
Rebased to remove commits for spelling and grammar fixes.

**Pull request merged**